src/Orleans.Core/Streams/Providers/IStreamProviderManager.cs

```diff
@@ -1,10 +0,0 @@
-using System.Collections.Generic;
-using Orleans.Providers;
-
-namespace Orleans.Streams
-{
-    public interface IStreamProviderManager : IProviderManager
-    {
-        IEnumerable<IStreamProvider> GetStreamProviders();
```

Comment:
why remove this public interface? or remove the ability to get all stream providers? we need to introduce as least as possible breaking change to legacy system, right?

Reply:
We can have a new implementation which wraps a `IServiceProvider`, which can easily give us all stream providers.

Reply:
We're removing the entire provider model, which is a breaking change, but 'should' not be bad as most users never see or use these surfaces.  The main thing we're keeping is the ability to load the custom providers which application developers have already developed.

Reply:
>We're removing the entire provider model

I guess I missed more than one meeting.. Do we still keep `IProvider` interface ,or is that removed too? can you share a big summary here about what removing entire provider model means?

Reply:
Yes, IProvider should eventually be removed (or moved to legacy).

To support extensibility comparable to the provider model, we need only the following:
*  Names services support (existing)
*  Generic silo lifecycle which services can take part in (existing)
*  Way for named services to take part in silo lifecycle (added in this PR), see usages of IKeyedServiceCollection<string, ILifecycleParticipant<IXXXClientLifecycle>>.

With these capabilities, orleans has no need for the provider model and related framework, including provider managers, and related abstractions (IProvider, IProviderConfiguration).  However, as framework extensions, users have developed custom providers, so we want to retain the ability to load those, which this PR also enables.